lib/std/jsonutils.nim

```diff
@@ -92,31 +98,59 @@ proc checkJsonImpl(cond: bool, condStr: string, msg = "") =
 template checkJson(cond: untyped, msg = "") =
   checkJsonImpl(cond, astToStr(cond), msg)
 
-template fromJsonFields(a, b, T, keys) =
-  checkJson b.kind == JObject, $(b.kind) # we could customize whether to allow JNull
+proc hasField[T](obj: T, field: string): bool =
+  for k, _ in fieldPairs(obj):
```

Comment:
`for k in fields(obj)` seems more efficient?

Reply:
Unfortunately, `fields` iterator yields objects values (only the second yield value of `fieldPairs`), but not the names of the fields (the first yield value of `fieldPairs`).